### PR TITLE
refactor: add webpack-defaults

### LINKED ISSUE #14: Wrong line number in Error
`require("line03")` is on line 3, but the error message tell us it is on line 5.

### PR DESCRIPTION
**Pull request by michael-ciniawsky**
### Fix

- [x] Error Message

### Test

- [x] Add Tests

### Docs

- [x] Add LOADER.md (JSDoc)
- [x] Update README.md

### Refactor

- [x] Add `webpack-defaults`
- [x] Support CoffeeScript v2
- [x] Load Sourcemaps only if setup (`options.sourceMap`)

### TODO 

- [ ] add `schema-utils`

Fixes #14, #21, #31, #32, #33

### CODE REVIEW

**Comment by michael-ciniawsky** [comment]
The ESLint Config is interpreted a bit tbh 😛 , but I would like to open discussion about a few of these rules anyways `import/order, import/no-dynamic-require, global-require, no-param-reassign`

File: test/webpack.build.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+const fs = require('fs-extra');
```
Comment (d3viant0ne) [inline test/webpack.build.js]: Personally, I'd just disable eslint for any test utility files, they have little to no bearing on code quality where it really matters.

Comment (d3viant0ne) [inline test/webpack.build.js]: To the point, the rules you mentioned in the comment above have an effect on code readability / quality for source files and are there for a reason imo.

For something like `webpack.build.js` where it has no effect on the above, I would sooner disable the linting outright in the file than disable rules across the organization that have a positive effect on overall quality.

Comment (d3viant0ne) [inline test/webpack.build.js]: `global-require` should not be disabled by default as it's frowned upon in general.

In the certain cases where it's necessary, it should be disabled inline on a case by case basis.

Comment (michael-ciniawsky) [inline test/webpack.build.js]: Still `import/order no-param-reassign` are 'conflicting' with the loader logic, the first for style/readability the latter for sourcemaps

```js
import fs from 'fs'
import path from 'path'

import { getOptions } from 'loader-utils'
import { validateOptions } from 'schema-utils'

import helper from './helper'
```

```js
if (options.sourceMap) {
  map = updateMap(map)
}
```

Comment (d3viant0ne) [inline test/webpack.build.js]: import/order - https://github.com/webpack-contrib/eslint-config-webpack/pull/7

File: package.json [hunk 2]
```diff
@@ -1,22 +1,73 @@
...
+    "webpack": "^2.3.3",
+    "webpack-defaults": "^0.4.5"
+  },
+  "engines": {
+    "node": ">= 4.3 < 5.0.0 || >= 5.10"
+  },
+  "main": "dist/cjs.js",
+  "files": [
+    "dist"
+  ],
```
Comment (michael-ciniawsky) [inline package.json]:
```
"repository": {
    "type": "git",
    "url": "git+https://github.com/webpack-contrib/coffee-loader.git"
  },
  "bugs": {
    "url": "https://github.com/webpack-contrib/coffee-loader/issues"
  },
  "homepage": "https://webpack.js.org",
```

**Comment by d3viant0ne** [comment]
@michael-ciniawsky - As we work through these, just create a named branch in the origin repo, makes it easier to test out for the group. You have org wide write access, far simpler than maintaining a fork :)

**Pull request closed without merging**